packages/gatsby-theme-vtex/src/sdk/newsletter/useNewsletter.ts

```diff
@@ -0,0 +1,66 @@
+import { useCallback, useState } from 'react'
+
+type MDResponse = {
+  Id: string
+  Href: string
+  DocumentId: string
+}
+
+const addToNewsletter = async (data: { name: string; email: string }) =>
+  fetch('/api/dataentities/NL/documents/', {
+    headers: {
+      'Content-Type': 'application/json',
+      accept: 'application/vnd.vtex.ds.v10+json',
+    },
+    method: 'PATCH',
+    body: JSON.stringify(data),
+  }).then((response) => {
+    if (response.ok) {
+      return response.json()
+    }
+
+    console.error(response)
+    throw new Error()
```

Comment:
What about adding some description to this error?

Reply:
Yeah, unfortunately we have no way to guarantee how our API errors come... That's why I've only thrown one Error and let the caller decide the error message.

Although it's highly unlikely that any error message from the Master Data would be directed to the user